### PR TITLE
Support Kubernetes 1.5

### PR DESCRIPTION
**Pull request by travisn**
Rook can run on Kubernetes 1.5 by using the yaml files in the 1.5 folder.

### CODE REVIEW

File: demo/kubernetes/1.5/mysql.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+apiVersion: v1
```
Comment (jbw976) [inline demo/kubernetes/1.5/mysql.yaml]: any of these files that are duplicated between 1.5 and the main demo/kubernetes folder should be symlinked instead, which will reduce the duplication.

File: Documentation/k8s-pre-reqs.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
 
-**WARNING:** Only Kubernetes v1.6.0 or higher is supported by Rook at this time (while Rook is in alpha it will track the latest release to use the latest features)
+## Minimum Version
+Kubernetes v1.6 or higher is targeted by Rook (while Rook is in alpha it will track the latest release to use the latest features). 
+
+Support is available for Kubernetes **v1.5.2**, although your mileage may vary. 
+You will need to use the yaml files from the [1.5 folder](/demo/kubernetes/1.5).
+
+## Privileges
+Creating the Rook operator requires privileges for setting up RBAC. To launch the operator you need to have created your user certificate with the `system:masters` privilege:
```
Comment (jbw976) [inline Documentation/k8s-pre-reqs.md]: is there a kubernetes doc link we can use here that approaches the general need for a cert?

**Comment by kokhang** [comment]
@travisn @jbw976 What is the main driver to support 1.5? And what kind of support would it be? I am scared that by "supporting" 1.5, we will be adding a new test matrix.

File: Documentation/k8s-pre-reqs.md (hunk 2 above)
Comment (travisn) [inline Documentation/k8s-pre-reqs.md]: can't seem to find an obvious simple link for this

File: demo/kubernetes/1.5/mysql.yaml (hunk 1 above)
Comment (travisn) [inline demo/kubernetes/1.5/mysql.yaml]: yes, good idea

**Comment by travisn** [comment]
@kokhang the motivation is that there are a lot of people still on 1.5 and they are blocked from using Rook. Support is a best effort and mileage may vary as added to the kubernetes.md doc.

**Comment by kokhang** [comment]
@travis I am wondering if it's better to handle these on a case by case basis. My concern is that if we add examples to rook to run on v1.5, this could mislead users into thinking that rook is supported on v1.5.

**Comment by travisn** [comment]
@kokhang the cost of supporting on a case by case basis will be much higher if we don't enable it with this change.

File: demo/kubernetes/1.5/rook-storageclass.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+metadata:
+   name: rook-block
+   namespace: rook
+provisioner: rook.io/block
+parameters:
+    pool: replicapool
+    # Specify the Rook cluster from which to create volumes. If not specified, it will use A command line client for working with a rook cluster.
+https://github.com/rook/rook
+
+Usage:
```
Comment (jbw976) [inline demo/kubernetes/1.5/rook-storageclass.yaml]: what is this rook usage output in the yaml file?

Comment (travisn) [inline demo/kubernetes/1.5/rook-storageclass.yaml]: that's odd, it's removed now

**Comment by debianmaster** [comment]
thank you very much !  i will test w/ openshift and let you know

**Comment by travisn** [comment]
ok, the build just completed so you can use this tag if you want to avoid master-latest: `v0.4.0-21-g5d619aa`

**Pull request merged**